silkworm/rpc/http/channel_writer.hpp

```diff
@@ -0,0 +1,56 @@
+/*
+   Copyright 2023 The Silkworm Authors
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+#pragma once
+
+namespace silkworm::rpc::http {
+
+class ChannelWriter {
+  public:
+    enum class ResponseStatus {
+        processing_continue,
+        accepted,
+        ok,
+        created,
+        no_content,
+        multiple_choices,
+        moved_permanently,
+        moved_temporarily,
+        not_modified,
+        bad_request,
+        unauthorized,
+        forbidden,
+        not_found,
+        internal_server_error,
+        not_implemented,
+        bad_gateway,
+        service_unavailable
+    };
+
+    struct MessageResponse {
```

Comment:
I suggest renaming `MessageResponse` as `Response`. Moreover, we can evaluate if this can be moved outside of the enclosing class but let's come back to this when this refactoring is finished.